AI/VCAI/VCAI.cpp

```diff
@@ -2668,6 +2667,7 @@ int3 VCAI::explorationNewPoint(HeroPtr h)
 	int radius = h->getSightRadius();
 	CCallback * cbp = cb.get();
 	const CGHeroInstance * hero = h.get();
+	const CPathsInfo * pathsInfo = cbp->getPathsInfo(hero);

```

Comment:
You should put `getPathsInfo` even upper on call stack, where `h` is set.

Reply:
We should not cache it - it invalidates very often (almost on any gamestate change). So you should put it to upper most function but do not store between AI API calls. (IOW put it so it transferred in a function parameter but do not stored in class field)